### ISSUE TITLE
bug: vue 3 vite-plugin-vue-layouts routing duplication

### ISSUE THREAD

**Issue by riderx** [ionic-team/ionic-framework#24192]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

To use tab component it's need to use `ionic/vue-router`
but i cannot use `router-view` anymore or i get duplicated route.
<img width="549" alt="Screenshot 2021-11-09 at 17 41 07" src="https://user-images.githubusercontent.com/4084527/140976286-606523c2-2c7e-4f3c-87e8-8853ad5162bb.png">
if i switch totally to `ion-router-outlet` i have css issue.
<img width="709" alt="Screenshot 2021-11-09 at 17 40 59" src="https://user-images.githubusercontent.com/4084527/140976308-50f673fd-bac8-4d1d-a9d5-db555d6c7c1e.png">

### Expected Behavior

Using ionic without ionic/vue-router should be allowed for tabs with `router-view` 
Or 
Allowing to have other route with `router-view`
Or
Ability to disabe host css for `ion-router-outlet`
<img width="323" alt="Screenshot 2021-11-09 at 16 08 30" src="https://user-images.githubusercontent.com/4084527/140976645-4c361b41-856c-4624-83fa-9840eb9cb7b4.png">

### Steps to Reproduce

clone this repo :
https://github.com/riderx/vitesse-ionic/tree/bug
do`pnpm i` 
do `pnpm dev`
go to the new server you should see the duplicated route
To see the second form of but write something in the input and click button `go`
then you will see footer component under hi page.
i found a workaround to make it work with `v-if` condition on `ion-router-outlet` as see there :
https://github.com/riderx/vitesse-ionic/blob/main/src/App.vue

### Code Reproduction URL

https://github.com/riderx/vitesse-ionic/tree/bug

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 6.0.0-rc.2

Utility:

   cordova-res (update available: 0.15.3) : 0.15.2
   native-run                             : 1.5.0

System:

   NodeJS : v14.18.1 (/usr/local/bin/node)
   npm    : 6.14.15
   OS     : macOS Monterey

### Additional Information

i made issue on package who have conflic with ionic :
the https://github.com/JohnCampionJr/vite-plugin-vue-layouts/issues/46

**Comment by liamdebeasi** [comment]
Thanks for the issue. Mixing `router-view` with `ion-router-outlet` is not something we support at the moment. Can you update your reproduction to show me the issue you are having when only using `ion-router-outlet`?

**Comment by riderx** [comment]
@liamdebeasi both exemple are present, the [hi page](https://github.com/riderx/vitesse-ionic/blob/bug/src/pages/hi/%5Bname%5D.vue) use the default [layout](https://github.com/riderx/vitesse-ionic/blob/bug/src/layouts/default.vue) who use only  `ion-router-outlet`

**Comment by liamdebeasi** [comment]
At first glance I can see that your "hi page" does not use the `IonPage` component. Any pages rendered inside of `ion-router-outlet` need to be wrapped in `<ion-page>` in order to work properly.

There's a lot going on in your app and I get the following error, so it is hard to debug: 

```
Module not found: Error: Can't resolve './src/main.js' in '/Users/liamdebeasi/Ionic/vitesse-ionic'
```

Can you try adding `<ion-page>` to the pages you render inside of `ion-router-outlet` and let me know if that solves the issue?

**Comment by riderx** [comment]
@liamdebeasi i have try already with `ion-page` i just added back.
For you issue do you use vscode recommended extension ? It's mendatory to have no error with vitesse

**Comment by liamdebeasi** [comment]
I still cannot run your application as I get the error mentioned in https://github.com/ionic-team/ionic-framework/issues/24192#issuecomment-969389993. It looks like it is expecting `main.js` to exist, but your project has `main.ts`. Can you please double check your reproduction case and ensure it contains only the minimum amount of code needed to reproduce the issue?

> For you issue do you use vscode recommended extension ? It's mendatory to have no error with vitesse

I don't use VSCode, so I cannot make any recommendations on extensions to use.

**Comment by riderx** [comment]
@liamdebeasi in the bug branch ?

**Comment by liamdebeasi** [comment]
Yes, I am using the `bug` branch.

**Comment by riderx** [comment]
Vitesse framework cannot be use without vscode i think, @antfu can you confirm?
 it has specific features that cannot be used without recommended extension.

**Comment by liamdebeasi** [comment]
Can you reproduce this issue without Vitesse?

**Comment by riderx** [comment]
@liamdebeasi the issue vanish when i remove `vite-plugin-vue-layouts` code so i think Vitesse is not the issue, do you need more isolated env with `vite-plugin-vue-layouts` only ?

**Comment by liamdebeasi** [comment]
Yes, if you could isolate the issue with only `vite-plugin-vue-layouts` and Ionic that would be great.

**Comment by riderx** [comment]
@liamdebeasi i found a way to reproduce it without anything more that vue 3, i think the issue is only in ionic with neasted route check it here:
https://codesandbox.io/s/vue-nested-routes-forked-9537y?file=/src/App.vue
all the nested view view is above the parent view:
<img width="721" alt="Screenshot 2021-11-16 at 16 07 04" src="https://user-images.githubusercontent.com/4084527/142021806-76101f33-7496-4b00-974f-6fa6869f2a04.png">
if you uncomment `router-view` in `app.vue` and comment the `ion-router-outlet` you will see the normal behavior:
<img width="721" alt="Screenshot 2021-11-16 at 16 07 15" src="https://user-images.githubusercontent.com/4084527/142021812-c9957cdb-1dfc-4af5-98a1-7fe6104975b5.png">

**Comment by liamdebeasi** [comment]
The issue here is that `ion-router-outlet` has `position: absolute`, so your page content is going to appear above the Vue logo. This is not a bug as the `position: absolute` is necessary in order for Ionic layout and page transitions to work properly. In general, we do not recommend putting content as a sibling of that root `ion-router-outlet`. Does that address your concern?

**Comment by riderx** [comment]
@liamdebeasi neasted route are made by `vite-plugin-vue-layouts` who is a part of Vitesse framework.

i would love to don't use`ionic/vue-router` but i cannot use **tab** component without it,
As i asked in the issue i need on the the 3 possible solution for me:

Allow using ionic without `ionic/vue-router` for tabs
Or
Allowing to have child with `router-view` instead of `ion-router-outlet`
Or
Ability to disabe/overload host css for ion-router-outlet in specific scenario

Do you think one of the 3 ways are doable ?
In my opinion `ionic/vue-router` could be optional, for tab, i don't use it in my app, basic routing is enought mostly

**Comment by liamdebeasi** [comment]
We built Ionic Vue with the intention that Ionic Vue Router would be used for developers looking to add routing support. Using `ion-tabs` without routing is not something we are looking to support at this time.

You can override the `ion-router-outlet` CSS for a specific use case, but you may run into other layout issues.

I think the main issue here is some confusion around how routing should and should not be customized inside of Ionic. We intended routing to be used for stack-based navigation where each view is contained in its own component with `ion-router-outlet` being given full control over how and when pages are displayed. I recommend reviewing the [Your First App Tutorial](https://ionicframework.com/docs/vue/your-first-app) which goes over the basics of how to use routing and tabs in Ionic Vue.

If you still have questions about how best to consume routing in Ionic, I strongly recommend you post on the [Ionic Forums](https://forum.ionicframework.com/) for feedback on your application. I am going to close this as there is nothing actionable from the Ionic Team.

**Comment by riderx** [comment]
@liamdebeasi neasting is a feature of `vue-router `as the doc says `we recommend relying on Vue Router features as much as possible rather than trying to build your own routing solutions.` https://ionicframework.com/docs/vue/navigation#ionrouteroutlet
I cannot do that, in that case.
In mobile i agree it can make sense to don't nest vue, but not in desktop.
I build app for both, that why i have choose ionic 6 years ago and i still believe, that ionic vision.

**Comment by liamdebeasi** [comment]
IonRouterOutlet is built on top of `vue-router`, so we are not building out own routing solution. In addition, `ion-router-outlet` supports nesting, but only with other `ion-router-outlet` components. See the [Nested Routes](https://ionicframework.com/docs/vue/navigation#shared-urls-versus-nested-routes) documentation for information on how to do this in Ionic.

**Comment by liamdebeasi** [comment]
Our Vue test app has an example of how to do nested outlets:

Routing config: https://github.com/ionic-team/ionic-framework/blob/main/packages/vue/test-app/src/router/index.ts#L61-L79

**Comment by riderx** [comment]
@liamdebeasi i understand your point about why we should use `ion-vue-router`, but it don't work in my scenario.

Please understand mine, using component and transition/animation do not seems corelated.
In user point of view, `ion-vue-router` is a separated package from `ionic/vue`,
 it's like you are saying i cannot use `vue` feature without `vue-router`. it doesn't make sence, otherwise it's a dependency or in the same module.

**Comment by liamdebeasi** [comment]
Can you please explain what you are trying to accomplish? From my perspective it sounds like you are trying to do nested routing, which is something that Ionic Vue supports.

You can use most components in Ionic Vue without ever installing Ionic Vue Router. There are some exceptions, such as IonRouterOutlet and IonTabs. We opted to keep these components in the Ionic Vue to make the developer experience more consistent. In other words, we did not want users to have to import components from two different libraries.

**Comment by riderx** [comment]
Yes, my usage is simple usually i build my app with the same stack i build the website and for that i need feature of vue-router like neasted route, my current framework depend on it Vitesse.
And because of this i cannot use in the mobile part the tab component.
Exemple here :
https://captime.app
the demo website and the app share the same code.

Since it wasn't difficult and it breaks nothing i made a PR even if you says it's not the way ionic is going, please consider this possibility maybe with a warning:
https://github.com/ionic-team/ionic-framework/pull/24224
